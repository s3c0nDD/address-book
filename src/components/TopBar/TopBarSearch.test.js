import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import TopBarSearch from './TopBarSearch';

describe('TopBarSearch', () => {
  let shallow;
  let component;
  let wrapper;
  let handleChangeMock;

  beforeAll(() => {
    shallow = createShallow();
    handleChangeMock = jest.fn();
  });

  beforeEach(() => {
    component = (
      <TopBarSearch
        disabled={false}
        text=""
        onTextChange={handleChangeMock}
      />
    );
    wrapper = shallow(component);
  });

  it('shallowly renders and matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
