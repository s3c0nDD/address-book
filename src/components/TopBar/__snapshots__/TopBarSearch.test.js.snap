// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBarSearch shallowly renders and matches snapshot 1`] = `
<div
  className="makeStyles-search-1"
>
  <div
    className="makeStyles-searchIcon-2"
  >
    <SearchIcon />
  </div>
  <WithStyles(ForwardRef(InputBase))
    classes={
      Object {
        "input": "makeStyles-inputInput-4",
        "root": "makeStyles-inputRoot-3",
      }
    }
    disabled={false}
    inputProps={
      Object {
        "aria-label": "search",
      }
    }
    onChange={[MockFunction]}
    placeholder="Search by full name"
    value=""
  />
</div>
`;
